src/lib/AppRegistry.tsx

```diff
@@ -88,14 +90,33 @@ const Conversation: React.SFC<ConversationProps> = track<ConversationProps>(prop
 
 const MyProfile: React.SFC<{}> = () => <MyProfileRenderer render={renderWithLoadProgress(Containers.MyProfile)} />
 
-interface BidFlowProps {
+type BidderFlowIntent = "bid" | "register"
+interface BidderFlowProps {
   artworkID?: string
   saleID: string
-  intent: "bid" | "register"
+  intent: BidderFlowIntent
+}
+
+interface BidderFlow {
+  queryRenderer: React.ComponentType<BidderFlowRendererProps>
+  container: React.ReactType<any>
+}
+
+const BidderFlows: { [BidderFlowIntent: string]: BidderFlow } = {
```

Comment:
Is it possible to add tests for this? (even just to make sure that when the intent is specified a certain way, the expected component shows)?

Reply:
Checking in with @orta on this.

Reply:
Hrm, off-hand I'm not even too sure what this does, might be worth some comments TBH as it's code that's being smart to avoid some duplication from the looks of it. 

This would be very, very tricky to test, probably not worth the effort given that it would just not load at dev time rather that break in a subtle way